pkg/e2e/verify/user_webhook.go

```diff
@@ -27,15 +27,390 @@ var _ = ginkgo.Describe(userWebhookTestName, func() {
 	h := helper.New()
 
 	ginkgo.Context("user validating webhook", func() {
-		ginkgo.It("dedicated admins cannot manage redhat users", func() {
+
+		//system:admin can do whatever it wants
+
+		ginkgo.It("system:admin can manage redhat users with SRE IDP and RH group", func() {
+			userName := util.RandomStr(5) + "@redhat.com"
+			identities := []string{"OpenShift_SRE:" + util.RandomStr(5)}
+			groups := []string{"osd-devaccess"}
+			// we need to add the username to the group before we create the user,
+			// because the user object cannot be created until the username is in
+			// the group
+			addUserToGroup(userName, groups[0], h)
+			h.Impersonate(rest.ImpersonationConfig{
+				UserName: "system:admin",
+				Groups: []string{
+					"system:authenticated",
+					"system:authenticated:oauth",
+				},
+			})
+			user, err := createUser(userName, identities, groups, h)
+			defer func() {
+				h.Impersonate(rest.ImpersonationConfig{})
```

Comment:
I'm unfamiliar with the defaults for an ImpersonationConfig, but why reset it here? Is it guaranteed that using this, the user can be deleted?

Reply:
yes, exactly - at the end of this test, we want to clean up our mess regardless of how the test went